genesyscloud/journey_outcome_predictor/genesyscloud_journey_outcome_predictor_proxy.go

```diff
@@ -0,0 +1,118 @@
+func newJourneyOutcomePredictorProxy(clientConfig *platformclientv2.Configuration) *journeyOutcomePredictorProxy {
+	journeyApi := platformclientv2.NewJourneyApiWithConfig(clientConfig)
+	return &journeyOutcomePredictorProxy{
+		clientConfig:                           clientConfig,
+		journeyApi:                             journeyApi,
+		createJourneyOutcomePredictorAttr:      createJourneyOutcomePredictorFn,
+		getAllJourneyOutcomePredictorAttr:      getAllJourneyOutcomePredictorFn,
+		getJourneyOutcomePredictorByIdAttr:     getJourneyOutcomePredictorByIdFn,
+		deleteJourneyOutcomePredictorAttr:      deleteJourneyOutcomePredictorFn,
+	}
+}
+
+// getJourneyOutcomePredictorProxy acts as a singleton to for the internalProxy.  It also ensures
+// that we can still proxy our tests by directly setting internalProxy package variable
+func getJourneyOutcomePredictorProxy(clientConfig *platformclientv2.Configuration) *journeyOutcomePredictorProxy {
+	if internalProxy == nil {
+		internalProxy = newJourneyOutcomePredictorProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// createJourneyOutcomePredictor creates a Genesys Cloud journey outcome predictor
+func (p *journeyOutcomePredictorProxy) createJourneyOutcomePredictor(ctx context.Context, outcomePredictor *platformclientv2.Outcomepredictorrequest) (*platformclientv2.Outcomepredictor, error) {
+	return p.createJourneyOutcomePredictorAttr(ctx, p, outcomePredictor)
+}
+
+// getJourneyOutcomePredictor retrieves all Genesys Cloud journey outcome predictor
+func (p *journeyOutcomePredictorProxy) getAllJourneyOutcomePredictor(ctx context.Context) (*[]platformclientv2.Outcomepredictor, error) {
+	return p.getAllJourneyOutcomePredictorAttr(ctx, p)
+}
+
+// getJourneyOutcomePredictorById returns a single Genesys Cloud journey outcome predictor by Id
+func (p *journeyOutcomePredictorProxy) getJourneyOutcomePredictorById(ctx context.Context, predictorId string) (journeyOutcomePredictor *platformclientv2.Outcomepredictor, statusCode int, err error) {
+	return p.getJourneyOutcomePredictorByIdAttr(ctx, p, predictorId)
+}
+
+// deleteJourneyOutcomePredictor deletes a Genesys Cloud journey outcome predictor by Id
+func (p *journeyOutcomePredictorProxy) deleteJourneyOutcomePredictor(ctx context.Context, predictorId string) (statusCode int, err error) {
+	return p.deleteJourneyOutcomePredictorAttr(ctx, p, predictorId)
+}
+
+// createJourneyOutcomePredictorFn is an implementation function for creating a Genesys Cloud journey outcome predictor
+func createJourneyOutcomePredictorFn(ctx context.Context, p *journeyOutcomePredictorProxy, outcomePredictor *platformclientv2.Outcomepredictorrequest) (*platformclientv2.Outcomepredictor, error) {
+
+	predictor, _, err :=  p.journeyApi.PostJourneyOutcomesPredictors(*outcomePredictor)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to predictor: %s", err)
+	}
```

Comment:
When I remove `nil` and just return the error alone I get failures when I run `make build`

`# terraform-provider-genesyscloud/genesyscloud/journey_outcome_predictor
genesyscloud/journey_outcome_predictor/genesyscloud_journey_outcome_predictor_proxy.go:82:10: not enough return values
	have (error)
	want (*platformclientv2.Outcomepredictor, error)
genesyscloud/journey_outcome_predictor/genesyscloud_journey_outcome_predictor_proxy.go:94:10: not enough return values
	have (error)
	want (*[]platformclientv2.Outcomepredictor, error)
genesyscloud/journey_outcome_predictor/genesyscloud_journey_outcome_predictor_proxy.go:108:27: not enough return values
	have (int, error)
	want (*platformclientv2.Outcomepredictor, int, error)
make: *** [build] Error 1`

Reply:
Sorry, I wasn't clear. I meant you could just `return nil, err` without the extra string info, since the function that invokes `createJourneyOutcomePredictorFn` will add that (line 62 in genesyscloud/journey_outcome_predictor/resource_genesyscloud_journey_outcome_predictor.go)